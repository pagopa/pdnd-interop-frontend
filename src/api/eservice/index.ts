export * from './eservice.queries'
export * from './eservice.mutations'
export * from './eservice.downloads'
export * from './eservice.services'
