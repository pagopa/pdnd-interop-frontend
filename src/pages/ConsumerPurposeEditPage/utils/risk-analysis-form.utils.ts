import type { Dependency, FormConfigQuestion, HideOption } from '@/api/api.generatedTypes'
import type { AnswerValue, Answers, Questions } from '../types/risk-analysis-form.types'

/**
 * Transform the answers from the backend to the frontend format
 * The answer from the backend arrives as an array of strings, but the frontend
 * needs a specific format depending on the `visualType` value.
 *
 * - `switch` needs a boolean
 * - `checkbox` needs an array of strings
 * - the rest needs a string
 *
 * @param backendAnswer The answer from the backend
 * @param question The question that the answer belongs to
 * @returns The answer in the type that the frontend needs
 */
export function getFrontendAnswerValue(
  backendAnswer: Array<string>,
  visualType?: FormConfigQuestion['visualType']
): AnswerValue {
  if (visualType === 'switch') {
    return backendAnswer[0] === 'true'
  }

  if (visualType === 'checkbox') {
    return backendAnswer
  }

  return backendAnswer[0]
}

/**
 * Transform the answers from the frontend to the backend format
 * The backend always needs an array of strings.
 *
 * @param answer The answer from the frontend
 * @returns The answer in the type that the backend needs
 * */
export function getBackendAnswerValue(answer: AnswerValue): Array<string> {
  if (typeof answer === 'boolean') {
    return [answer.toString()]
  }

  if (Array.isArray(answer)) {
    return answer
  }

  if (typeof answer === 'string') {
    return [answer]
  }

  throw new Error(`Invalid answer type: ${typeof answer} \n\n ${JSON.stringify(answer, null, 2)}`)
}

/**
 * Returns the valid answers that should be sent to the backend.
 * The answers that should be sent to the backend are only the answers
 * to the questions that are currently visible to the user.
 * The answers to the questions that are not visible to the user are not sent.
 *
 * @param currentQuestionsIds - the ids of the questions that are currently visible to the user
 * @param answers - the answers to all the questions
 * @returns the valid answers that should be sent to the backend
 * */
export function getValidAnswers(currentQuestionsIds: Array<string>, answers: Answers) {
  return Object.keys(answers).reduce<Record<string, Array<string>>>((acc, key) => {
    // If the question is actually visible, add it to the valid answers
    if (currentQuestionsIds.includes(key)) {
      return { ...acc, [key]: getBackendAnswerValue(answers[key]) }
    }
    return acc
  }, {})
}

/**
 * Returns true if the dependency is satisfied, false otherwise.
 *
 * @param dependency - the dependency
 * @param answers - the actual form values
 * @returns `true` if the dependency is satisfied, `false` otherwise
 * */
export function isDependencySatisfied(dependency: Dependency | HideOption, answers: Answers) {
  const answer = answers[dependency.id]
  if (Array.isArray(answer)) {
    return answer.includes(dependency.value)
  }

  return answer === dependency.value
}

/**
 * Returns the updated question data.
 *
 * @param answers - the actual form values
 * @param riskAnalysis - the risk analysis document
 *
 * @returns the updated questions data
 * */
export function getUpdatedQuestions(
  answers: Answers,
  riskAnalysisQuestions: FormConfigQuestion[]
): Questions {
  // Filters all the questions that not satisfies the dependency inside the "dependencies" property
  return riskAnalysisQuestions.reduce<Questions>((acc, question) => {
    const doesSatisfyDeps = question.dependencies.every((dependency) =>
      isDependencySatisfied(dependency, answers)
    )
    if (doesSatisfyDeps) {
      acc[question.id] = question
    }
    return acc
  }, {})
}

/**
 * Returns the default values for the risk analysis form.
 * The default values are the values that the form should have when it is first loaded.
 * The default values are the values that the user has already answered, if any.
 * If the user has not answered any question, the values are the default values
 * defined in the risk analysis document.
 *
 * @param riskAnalysisConfig the risk analysis document
 * @param backendAnswers the answers data, taken from the Purpose
 * @returns the default values for the risk analysis form
 */
export function getRiskAnalysisDefaultValues(
  riskAnalysisConfigQuestions: FormConfigQuestion[],
  backendAnswers?: Record<string, Array<string>>
): Answers {
  return riskAnalysisConfigQuestions.reduce<Answers>((acc, question) => {
    const answer = backendAnswers?.[question.id] ?? question.defaultValue
    acc[question.id] = getFrontendAnswerValue(answer, question.visualType)

    if (!acc[question.id] && question.dataType === 'FREETEXT') {
      acc[question.id] = ''
    }

    return acc
  }, {})
}
