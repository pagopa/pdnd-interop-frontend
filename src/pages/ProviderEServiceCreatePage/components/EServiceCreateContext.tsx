import React from 'react'
import { createContext } from '@/utils/common.utils'
import noop from 'lodash/noop'
import type { ProducerEServiceDescriptor, ProducerEServiceDetails } from '@/api/api.generatedTypes'

type EServiceCreateContextType = {
  eservice: ProducerEServiceDetails | ProducerEServiceDescriptor['eservice'] | undefined
  descriptor: ProducerEServiceDescriptor | undefined
  isNewEService: boolean
  back: VoidFunction
  forward: VoidFunction
}

const initialState: EServiceCreateContextType = {
  eservice: undefined,
  descriptor: undefined,
  isNewEService: false,
  back: noop,
  forward: noop,
}

const { useContext, Provider } = createContext<EServiceCreateContextType>(
  'EServiceCreateContext',
  initialState
)

type EServiceCreateContextProviderProps = {
  children: React.ReactNode
  eservice: ProducerEServiceDetails | ProducerEServiceDescriptor['eservice'] | undefined
  descriptor: ProducerEServiceDescriptor | undefined
  isNewEService: boolean
  back: VoidFunction
  forward: VoidFunction
}

const EServiceCreateContextProvider: React.FC<EServiceCreateContextProviderProps> = ({
  children,
  eservice,
  descriptor,
  isNewEService,
  back,
  forward,
}) => {
  const providerValue = React.useMemo(() => {
    return { eservice, descriptor, isNewEService, back, forward }
  }, [eservice, descriptor, isNewEService, back, forward])

  return <Provider value={providerValue}>{children}</Provider>
}

export { useContext as useEServiceCreateContext, EServiceCreateContextProvider }
