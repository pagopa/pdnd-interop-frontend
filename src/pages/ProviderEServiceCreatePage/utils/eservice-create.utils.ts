import type {
  AgreementApprovalPolicy,
  DescriptorAttributes,
  ProducerEServiceDescriptor,
} from '@/api/api.generatedTypes'
import type { RemappedEServiceAttributes } from '@/types/attribute.types'
import { getKeys } from '@/utils/array.utils'
import { remapEServiceAttributes } from '@/utils/attribute.utils'
import isEqual from 'lodash/isEqual'

type DescriptorDataStep2ToCompare = {
  voucherLifespan: number
  audience: string[]
  agreementApprovalPolicy: AgreementApprovalPolicy
  version: string
  description: string
  dailyCallsPerConsumer: number
  dailyCallsTotal: number
}

export function compareDescriptorsStep2DataIfEquals(
  newDescriptorData: DescriptorDataStep2ToCompare,
  descriptorToCompare: ProducerEServiceDescriptor
) {
  const descriptorDataToCompare: DescriptorDataStep2ToCompare = {
    voucherLifespan: descriptorToCompare.voucherLifespan,
    audience: descriptorToCompare.audience,
    agreementApprovalPolicy: descriptorToCompare.agreementApprovalPolicy,
    version: descriptorToCompare.version,
    description: descriptorToCompare.description ?? '',
    dailyCallsPerConsumer: descriptorToCompare.dailyCallsPerConsumer,
    dailyCallsTotal: descriptorToCompare.dailyCallsTotal,
  }

  return getKeys(newDescriptorData).every((key) =>
    isEqual(newDescriptorData[key], descriptorDataToCompare[key])
  )
}

export function compareAttributesStep3IfEquals(
  newAttributes: RemappedEServiceAttributes,
  attributesToCompare: DescriptorAttributes
) {
  const remappedAttributesToCompare = remapEServiceAttributes(attributesToCompare)

  const newAttributesToCompare = {
    certified: newAttributes.certified.filter((group) => group.attributes.length > 0),
    verified: newAttributes.verified.filter((group) => group.attributes.length > 0),
    declared: newAttributes.declared.filter((group) => group.attributes.length > 0),
  }

  return getKeys(newAttributes).every((key) =>
    isEqual(newAttributesToCompare[key], remappedAttributesToCompare[key])
  )
}
