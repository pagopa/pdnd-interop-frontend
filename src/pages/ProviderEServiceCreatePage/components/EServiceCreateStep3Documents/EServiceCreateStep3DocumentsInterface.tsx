import React from 'react'
import UploadFileIcon from '@mui/icons-material/UploadFile'
import { useTranslation } from 'react-i18next'
import { Stack, Box, Button } from '@mui/material'
import { useEServiceCreateContext } from '../EServiceCreateContext'
import { DocumentContainer } from '@/components/layout/containers/DocumentContainer'
import { FormProvider, useForm } from 'react-hook-form'
import { RHFSingleFileInput, RHFTextField } from '@/components/shared/react-hook-form-inputs'
import { EServiceDownloads, EServiceMutations } from '@/api/eservice'
import { getDownloadDocumentName } from '@/utils/eservice.utils'
import type { EServiceDoc } from '@/api/api.generatedTypes'

type EServiceCreateStep3DocumentsInterfaceFormValues = {
  interfaceDoc: File | null
  prettyName: string
}

export function EServiceCreateStep3DocumentsInterface() {
  const { t } = useTranslation('eservice')
  const { descriptor } = useEServiceCreateContext()
  const downloadDocument = EServiceDownloads.useDownloadVersionDocument()
  const { mutate: deleteDocument } = EServiceMutations.useDeleteVersionDraftDocument()
  const { mutate: uploadDocument } = EServiceMutations.usePostVersionDraftDocument()

  const defaultValues: EServiceCreateStep3DocumentsInterfaceFormValues = {
    interfaceDoc: null,
    prettyName: t('create.step3.interface.prettyName'),
  }

  const actualInterface: EServiceDoc | null = descriptor?.interface ?? null

  const formMethods = useForm({
    defaultValues,
    shouldUnregister: true,
  })

  const onSubmit = ({
    interfaceDoc,
    prettyName,
  }: EServiceCreateStep3DocumentsInterfaceFormValues) => {
    if (!interfaceDoc || !descriptor) return
    uploadDocument({
      eserviceId: descriptor.eservice.id,
      descriptorId: descriptor.id,
      doc: interfaceDoc,
      prettyName,
      kind: 'INTERFACE',
    })
  }

  const handleDeleteInterface = () => {
    if (!actualInterface || !descriptor) return
    deleteDocument({
      eserviceId: descriptor.eservice.id,
      descriptorId: descriptor.id,
      documentId: actualInterface.id,
    })
  }

  const handleDownloadInterface = () => {
    if (!actualInterface || !descriptor) return
    downloadDocument(
      {
        eserviceId: descriptor.eservice.id,
        descriptorId: descriptor.id,
        documentId: actualInterface.id,
      },
      getDownloadDocumentName(actualInterface)
    )
  }

  if (actualInterface) {
    return (
      <DocumentContainer
        doc={actualInterface}
        onDelete={handleDeleteInterface}
        onDownload={handleDownloadInterface}
      />
    )
  }

  return (
    <FormProvider {...formMethods}>
      <Box
        component="form"
        noValidate
        onSubmit={formMethods.handleSubmit(onSubmit)}
        sx={{ px: 2, py: 2, borderLeft: 4, borderColor: 'primary.main' }}
        bgcolor="common.white"
      >
        <RHFSingleFileInput
          sx={{ my: 0 }}
          name="interfaceDoc"
          label={t('create.step3.uploadFileField.label')}
          rules={{ required: true }}
        />

        <RHFTextField
          sx={{ my: 2 }}
          name="prettyName"
          label={t('create.step3.nameField.label')}
          disabled
          rules={{ required: true }}
        />

        <Stack direction="row" justifyContent="flex-end">
          <Button type="submit" variant="contained">
            <UploadFileIcon fontSize="small" sx={{ mr: 1 }} /> {t('create.step3.uploadBtn')}
          </Button>
        </Stack>
      </Box>
    </FormProvider>
  )
}
