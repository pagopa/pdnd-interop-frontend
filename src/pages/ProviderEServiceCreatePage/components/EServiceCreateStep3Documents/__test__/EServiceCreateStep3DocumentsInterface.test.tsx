import React from 'react'
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import { vi } from 'vitest'
import * as EServiceCreateContext from '../../EServiceCreateContext'
import type { EServiceDoc, ProducerEServiceDescriptor } from '@/api/api.generatedTypes'
import { renderWithApplicationContext } from '@/utils/testing.utils'
import { createMockEServiceDescriptorProvider } from '__mocks__/data/eservice.mocks'
import userEvent from '@testing-library/user-event'
import { BACKEND_FOR_FRONTEND_URL } from '@/config/env'
import { waitFor } from '@testing-library/react'
import { EServiceCreateStep3DocumentsInterface } from '../EServiceCreateStep3DocumentsInterface'
import { EServiceDownloads, EServiceMutations } from '@/api/eservice'

const docMock: EServiceDoc = {
  id: '1',
  name: 'name',
  contentType: 'pdf',
  prettyName: 'document',
}

const server = setupServer(
  rest.delete(
    `${BACKEND_FOR_FRONTEND_URL}/eservices/:eserviceId/descriptors/:descriptorId/documents/:documentId`,
    (req, res, ctx) => {
      return res(ctx.status(200))
    }
  ),
  rest.get(
    `${BACKEND_FOR_FRONTEND_URL}/eservices/:eserviceId/descriptors/:descriptorId/documents/:documentId`,
    (req, res, ctx) => {
      return res(ctx.status(200))
    }
  ),
  rest.post(
    `${BACKEND_FOR_FRONTEND_URL}/eservices/:eserviceId/descriptors/:descriptorId/documents`,
    (req, res, ctx) => {
      return res(ctx.status(200))
    }
  )
)

const original = window.crypto.getRandomValues
// eslint-disable-next-line @typescript-eslint/no-explicit-any
window.crypto.getRandomValues = () => [0, undefined] as any

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => {
  server.close()
  window.crypto.getRandomValues = original
})

const mockUseEServiceCreateContext = (descriptor: ProducerEServiceDescriptor | undefined) => {
  vi.spyOn(EServiceCreateContext, 'useEServiceCreateContext').mockReturnValue({
    descriptor,
  } as unknown as ReturnType<typeof EServiceCreateContext.useEServiceCreateContext>)
}

describe('EServiceCreateStep3DocumentsDoc', () => {
  it('should match snapshot if there is no actualInterface', () => {
    const descriptor = createMockEServiceDescriptorProvider({ interface: undefined })
    mockUseEServiceCreateContext(descriptor)

    const { baseElement } = renderWithApplicationContext(
      <EServiceCreateStep3DocumentsInterface />,
      {
        withReactQueryContext: true,
      }
    )

    expect(baseElement).toMatchSnapshot()
  })

  it('should match snapshot if there is actualInterface', () => {
    const descriptor = createMockEServiceDescriptorProvider({ interface: docMock })
    mockUseEServiceCreateContext(descriptor)

    const { baseElement } = renderWithApplicationContext(
      <EServiceCreateStep3DocumentsInterface />,
      {
        withReactQueryContext: true,
      }
    )

    expect(baseElement).toMatchSnapshot()
  })

  it('should call deleteDocument mutation', async () => {
    const descriptor = createMockEServiceDescriptorProvider({ interface: docMock })
    mockUseEServiceCreateContext(descriptor)

    const deleteVersionDraftDocumentFn = vi.fn()
    vi.spyOn(EServiceMutations, 'useDeleteVersionDraftDocument').mockReturnValue({
      mutate: deleteVersionDraftDocumentFn,
    } as unknown as ReturnType<typeof EServiceMutations.useDeleteVersionDraftDocument>)

    const screen = renderWithApplicationContext(<EServiceCreateStep3DocumentsInterface />, {
      withReactQueryContext: true,
    })

    const user = userEvent.setup()
    const deleteButton = screen.getByRole('button', { name: 'deleteDocument' })
    await user.click(deleteButton)

    await waitFor(() =>
      expect(deleteVersionDraftDocumentFn).toBeCalledWith({
        eserviceId: descriptor.eservice.id,
        descriptorId: descriptor.id,
        documentId: docMock.id,
      })
    )
  })

  it('should call downloadDocument', async () => {
    const descriptor = createMockEServiceDescriptorProvider({ interface: docMock })
    mockUseEServiceCreateContext(descriptor)

    const downloadDocumentFn = vi.fn()
    vi.spyOn(EServiceDownloads, 'useDownloadVersionDocument').mockReturnValue(
      downloadDocumentFn as unknown as ReturnType<
        typeof EServiceDownloads.useDownloadVersionDocument
      >
    )

    const screen = renderWithApplicationContext(<EServiceCreateStep3DocumentsInterface />, {
      withReactQueryContext: true,
    })

    const user = userEvent.setup()
    const downloadButton = screen.getByRole('button', { name: 'downloadDocument' })
    await user.click(downloadButton)

    await waitFor(() =>
      expect(downloadDocumentFn).toBeCalledWith(
        {
          eserviceId: descriptor.eservice.id,
          descriptorId: descriptor.id,
          documentId: docMock.id,
        },
        'document.name' // getDownloadDocumentName(docMock)
      )
    )
  })

  it('should call uploadDocument mutation', async () => {
    const descriptor = createMockEServiceDescriptorProvider({ interface: undefined })
    mockUseEServiceCreateContext(descriptor)

    const postVersionDraftDocumentFn = vi.fn()
    vi.spyOn(EServiceMutations, 'usePostVersionDraftDocument').mockReturnValue({
      mutate: postVersionDraftDocumentFn,
    } as unknown as ReturnType<typeof EServiceMutations.usePostVersionDraftDocument>)

    const screen = renderWithApplicationContext(<EServiceCreateStep3DocumentsInterface />, {
      withReactQueryContext: true,
    })

    const user = userEvent.setup()

    // const docInput = screen.getByLabelText('create.step3.uploadFileField.label')
    const docInput = screen.getByLabelText('')
    const testFile = new File(['file-test'], 'test.pdf', { type: 'pdf' })
    await user.upload(docInput, testFile)

    const submitButton = screen.getByRole('button', { name: 'create.step3.uploadBtn' })
    await user.click(submitButton)

    await waitFor(() =>
      expect(postVersionDraftDocumentFn).toBeCalledWith({
        eserviceId: descriptor.eservice.id,
        descriptorId: descriptor.id,
        doc: testFile,
        prettyName: 'create.step3.interface.prettyName',
        kind: 'INTERFACE',
      })
    )
  })

  it('should skip uploadDocument mutation if there is no descriptor', async () => {
    mockUseEServiceCreateContext(undefined)

    const postVersionDraftDocumentFn = vi.fn()
    vi.spyOn(EServiceMutations, 'usePostVersionDraftDocument').mockReturnValue({
      mutate: postVersionDraftDocumentFn,
    } as unknown as ReturnType<typeof EServiceMutations.usePostVersionDraftDocument>)

    const screen = renderWithApplicationContext(<EServiceCreateStep3DocumentsInterface />, {
      withReactQueryContext: true,
    })

    const user = userEvent.setup()

    // const docInput = screen.getByLabelText('create.step3.uploadFileField.label')
    const docInput = screen.getByLabelText('')
    const testFile = new File(['file-test'], 'test.pdf', { type: 'pdf' })
    await user.upload(docInput, testFile)

    const submitButton = screen.getByRole('button', { name: 'create.step3.uploadBtn' })
    await user.click(submitButton)

    await waitFor(() => expect(postVersionDraftDocumentFn).not.toBeCalled())
  })

  it('should skip uploadDocument mutation if there is no interfaceDoc uploaded', async () => {
    const descriptor = createMockEServiceDescriptorProvider({ interface: undefined })
    mockUseEServiceCreateContext(descriptor)

    const postVersionDraftDocumentFn = vi.fn()
    vi.spyOn(EServiceMutations, 'usePostVersionDraftDocument').mockReturnValue({
      mutate: postVersionDraftDocumentFn,
    } as unknown as ReturnType<typeof EServiceMutations.usePostVersionDraftDocument>)

    const screen = renderWithApplicationContext(<EServiceCreateStep3DocumentsInterface />, {
      withReactQueryContext: true,
    })

    const user = userEvent.setup()
    const submitButton = screen.getByRole('button', { name: 'create.step3.uploadBtn' })
    await user.click(submitButton)

    await waitFor(() => expect(postVersionDraftDocumentFn).not.toBeCalled())
  })
})
