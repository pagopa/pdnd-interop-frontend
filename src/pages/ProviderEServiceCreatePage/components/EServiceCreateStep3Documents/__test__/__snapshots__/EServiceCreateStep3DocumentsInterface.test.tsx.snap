// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EServiceCreateStep3DocumentsDoc > should match snapshot if there is actualInterface 1`] = `
<body>
  <div>
    <div
      class="css-gmwslw-MuiStack-root"
    >
      <div
        class="MuiBox-root css-n2ek5d"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-8rphz-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r2:"
            id=":r2:-label"
          >
            prettyName.label
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                data-testid="AttachFileIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id=":r2:"
              maxlength="60"
              name="prettyName"
              type="text"
              value="document"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  prettyName.label
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="css-qkqpx0-MuiStack-root"
      >
        <button
          aria-label="downloadDocument"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1carnhg-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="DownloadIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="deleteDocument"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-15suzu7-MuiButtonBase-root-MuiButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="DeleteOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`EServiceCreateStep3DocumentsDoc > should match snapshot if there is no actualInterface 1`] = `
<body>
  <div>
    <form
      class="MuiBox-root css-1qmoj6l"
      novalidate=""
    >
      <div
        class="MuiBox-root css-1mgrf5c"
      >
        <div
          class="MuiFormControl-root css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1k551yo-MuiFormLabel-root"
            for="00000000-0000-4000-8000-000000000000"
          />
          <div
            class="MuiBox-root css-1alczda"
          >
            <button
              class="MuiBox-root css-1734zrc"
              data-testid="loadFromPc"
              type="button"
            >
              <div
                class="MuiBox-root css-1b8v44g"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-p2cdt4-MuiSvgIcon-root"
                  data-testid="CloudUploadIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-141vi1f-MuiTypography-root"
                >
                  dropzoneLabel
                </p>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1s3b67t-MuiInputBase-root-MuiInput-root"
                  data-testid="fileInput"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="00000000-0000-4000-8000-000000000000"
                    type="file"
                    value=""
                  />
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            create.step3.nameField.label
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id=":r0:"
              name="prettyName"
              type="text"
              value="create.step3.interface.prettyName"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  create.step3.nameField.label
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="css-181c0ax-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-15ikgma-MuiSvgIcon-root"
            data-testid="UploadFileIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11zM8 15.01l1.41 1.41L11 14.84V19h2v-4.16l1.59 1.59L16 15.01 12.01 11z"
            />
          </svg>
           
          create.step3.uploadBtn
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</body>
`;
